### PR TITLE
Fix the Kendalls Tau metric when used in graph mode

### PR DESCRIPTION
**Pull request by nicolaspi**
# Description
~Blocked by https://github.com/tensorflow/addons/pull/2740~

Brief Description of the PR:

The metric was only working in eager mode. This fix makes it work in graph mode too.

## Type of change

- [X] Bug fix

# Checklist:

- [X] I've properly [formatted my code according to the guidelines](https://github.com/tensorflow/addons/blob/master/CONTRIBUTING.md#coding-style)
    - [ ] By running Black + Flake8
    - [X] By running pre-commit hooks
- [ ] This PR addresses an already submitted issue for TensorFlow Addons
- [ ] I have made corresponding changes to the documentation
- [X] I have added tests that prove my fix is effective or that my feature works
- [ ] This PR contains modifications to C++ custom-ops

# How Has This Been Tested?

The metric was giving NaNs when running in graph mode. I modified the test to check against the NaN values.

### CODE REVIEW

**Review by sorensenjs — APPROVED** [review]
Looks great - thanks for this change.

**Review by sorensenjs — DISMISSED** [review]

**Comment by nicolaspi** [comment]
I removed a padding op that seemed unnecessary.

File: tensorflow_addons/metrics/kendalls_tau.py [hunk 1]
```diff
@@ -15,7 +15,7 @@
 """Approximate Kendall's Tau-b Metric."""
 
 import tensorflow as tf
-from tensorflow.keras.metrics import Metric
+from keras.metrics import Metric
```
Comment (bhack) [inline tensorflow_addons/metrics/kendalls_tau.py:18]: We  need to import Keras from TF

**Review by sorensenjs — APPROVED** [review]

**Comment by bhack** [comment]
@theadactyl Just another TF ecosystem duplication reminder, we have this also in TFP:
https://github.com/tensorflow/probability/issues/1417

**Comment by sorensenjs** [comment]
> @theadactyl Just another TF ecosystem duplication reminder, we have this also in TFP: [tensorflow/probability#1417](https://github.com/tensorflow/probability/issues/1417)

This is incorrect - the tfp implementation (1) can't be used as a keras metric (2) is not approximate. I'm the author of both, don't blame me for the complexities of TF packaging.

This particular implementation is an online O(n) approximate algorithm that is much better suited for use as a metric in machine learning. The TFP implementation is an O(n^2) algorithm.

**Comment by nicolaspi** [comment]
The failing check looks to come from a regression in tensorflow_docs https://github.com/tensorflow/docs/commit/eaef2e3cfe61d3940846d00ba022b44db6c96996 ?

**Comment by bhack** [comment]
> The failing check looks to come from a regression in tensorflow_docs https://github.com/tensorflow/docs/commit/eaef2e3cfe61d3940846d00ba022b44db6c96996 ?

/cc @MarkDaoust

**Comment by bhack** [comment]
> > @theadactyl Just another TF ecosystem duplication reminder, we have this also in TFP: [tensorflow/probability#1417](https://github.com/tensorflow/probability/issues/1417)
> 
> This is incorrect - the tfp implementation (1) can't be used as a keras metric (2) is not approximate. I'm the author of both, don't blame me for the complexities of TF packaging.
> 
> This particular implementation is an online O(n) approximate algorithm that is much better suited for use as a metric in machine learning. The TFP implementation is an O(n^2) algorithm.

I was exactly referring to the ecosystem boundaries and packaging.

See our original 2020 thread:
https://github.com/tensorflow/addons/pull/2169#issuecomment-696880578

**Review by bhack — APPROVED** [review]
